low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +45,129 @@ class ServerMap(ResourceMap):
+        mem_path = utility_instance.get_proc_memory('meminfo')
+        if mem_path.is_file():
+            meminfo_dict = self.read_mem_file(mem_path)
+            total_mem = meminfo_dict['MemTotal:']
+            path_exists, _ = self.store.exists(self.MEM_FAULT_SENSOR_DATA)
+            if path_exists:
+                self.get_stored_mem_info()
+                if self.fault_alert_state == "fault":
+                    status = "FAULT"
+                    description  = \
+                    "Total available main memory value decreased"\
+                    f"from {self.prev_mem} kB to {total_mem} kB"
+            memory_dict = self.prepare_mem_json(status,
+                                                description)
+            data.append(memory_dict)
+        else:
+            raise HealthProviderError(
+                errno.EINVAL,
+                f"{mem_path}: file does not exist")
+        return data
+
+    def read_mem_file(self, mem_path):
+        """ Get Memory information from /proc/meminfo."""
+        meminfo_dict = {}
+        self.mem_path_file = mem_path.read_text()
+        # populate all the data from /proc/meminfo
+        split_strs = [s.split(maxsplit=1) for s in self.mem_path_file.splitlines()]
+        meminfo_dict = dict(split_strs)
+        return meminfo_dict
+
+    def get_stored_mem_info(self):
+        mem_data = (self.store.get(self.MEM_FAULT_SENSOR_DATA)).split(":")
+        self.prev_mem = mem_data[0].strip()
+        self.fault_alert_state = mem_data[1].strip()
 
+    def prepare_mem_json(self, status, description):
+        total_memory = dict(psutil.virtual_memory()._asdict())
+        memory_dict = {
+            "uid": "memory_0",
+            "fru": "false",
+            "last_updated": int(time.time()),
+            "health": {
+            "status": status,
+            "description": description,
+            "specifics": [
+                {
+                   "total": total_memory['total'],
+                   "available": total_memory['available'],
+                   "percent": total_memory['percent'],
+                   "used": total_memory['used'],
+                   "free": total_memory['free'],
+                   "active": total_memory['active'],
+                   "inactive": total_memory['inactive'],
+                   "buffers": total_memory['buffers'],
+                   "cached": total_memory['cached'],
+                   "shared": total_memory['shared'],
+                   "slab": total_memory['slab']
+                }
+                ]
+            }
+        }
+        return memory_dict
+        
 
-# if __name__ == "__main__":
-#     server = ServerMap()
-#     health_data = server.get_health_info(rpath="nodes[0]>compute[0]>hw>disks")
-#     print(health_data)
+if __name__ == "__main__":
+    server = ServerMap()
+    health_data = server.get_health_info(rpath="nodes[0]>compute[0]>hw>memory")
+    print(health_data)
```

Comment:
These lines should be commented.
This module can be executed by any script in command line by any chance. So avoid main will be executed.

Reply:
Done.
just for testing purpose, added the code. Commented it now.